contracts/exchange/contracts/src/MixinSignatureValidator.sol

```diff
@@ -305,6 +350,53 @@ contract MixinSignatureValidator is
         ));
     }
 
+    /// @dev Verifies signature using logic defined by an EIP1271 Wallet contract.
+    /// @param hash Any 32 byte hash.
+    /// @param walletAddress Address that should have signed the given hash
+    ///                      and defines its own signature verification method.
+    /// @param signature Proof that the hash has been signed by signer.
+    /// @return True if the signature is validated by the Walidator.
+    function _validateHashWithEIP1271Wallet(
+        bytes32 hash,
+        address walletAddress,
+        bytes memory signature
+    )
+        private
+        view
+        returns (bool isValid)
+    {
+        uint256 signatureLength = signature.length;
+        // Shave the signature type off the signature.
+        assembly {
+            mstore(signature, sub(signatureLength, 1))
+        }
+        // Encode the call data.
+        bytes memory data = new bytes(32);
+        data.writeBytes32(0, hash);
+        bytes memory callData = abi.encodeWithSelector(
+            IEIP1271Wallet(walletAddress).isValidSignature.selector,
+            data,
+            signature
+        );
+        // Restore the full signature.
```

Comment:
I'm concerned that we may extend/refactor this someday, and it's not necessarily the case that this byte will still be in-place in memory. We could probably copy the logic from `LibAddressArray.append` to write a proper  `LibBytes.append`. Although if that's too expensive, perhaps just tag this as a hack and note not to write to memory.

Reply:
Yeah, `append` would be cleaner but feels like overkill for this, especially now that we verify certain signatures on every fill. The rest of `MixinSignatureValidator` has been previously rewritten to be non-destructive to the signature (mostly so we can revert with the full signature), but I will underscore the necessity for this pattern in the comments.